### PR TITLE
New tests for provideSequencerStaking

### PR DESCRIPTION
**Pull request by AlexChetverov**

### CODE REVIEW

File: test/sequential/sequencerStaking.sequencerJoining.test.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  );
+  const eventJoining = events.filter(
+    (x) => x.method === "SequencerJoinedActiveSet",
+  );
+  const eventReserved = events.filter((x) => x.method === "Reserved");
+  expect(eventJoining[0]).toBeUndefined();
+  expect(eventReserved[0].event.data[1].toHuman()).toContain(
+    testUser.keyRingPair.address,
+  );
+  expect(eventReserved[0].event.data[2]).bnEqual(stakeAmount);
```
Comment (goncer) [inline test/sequential/sequencerStaking.sequencerJoining.test.ts:53]: I think you can extract this to a utility method, so the logic about the event data  & indexes party are in a single place.

File: test/sequential/sequencerStaking.sequencerJoining.test.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+});
+
+it("GIVEN User provides a stake by using StakeAndJoinActiveSet action THEN User is a sequencer", async () => {
+  const events = await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser.keyRingPair.address,
+      await SequencerStaking.provideSequencerStaking(
+        stakeAmount,
+        chain,
+        "StakeAndJoinActiveSet",
```
Comment (goncer) [inline test/sequential/sequencerStaking.sequencerJoining.test.ts:67]: This and the other option deserves an enum or constant in Rolldown class I think

File: test/sequential/sequencerStaking.sequencerJoining.test.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    testUser.keyRingPair.address,
+  );
+  expect(eventReserved[0].event.data[2]).bnEqual(stakeAmount);
+  const sequencersList = await SequencerStaking.activeSequencers();
+  expect(sequencersList.toHuman().Ethereum).not.toContain(
+    testUser.keyRingPair.address,
+  );
+});
+
+it("GIVEN User provides a stake by using StakeAndJoinActiveSet action THEN User is a sequencer", async () => {
```
Comment (goncer) [inline test/sequential/sequencerStaking.sequencerJoining.test.ts:60]: what if using StakeAndJoinActiveSet but stake < minimum? could you please add this test?

File: test/sequential/sequencerStaking.sequencerJoining.test.ts [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        stakeAmount,
+        chain,
+        "StakeOnly",
+      ),
+    ),
+  );
+  const eventJoining = events.filter(
+    (x) => x.method === "SequencerJoinedActiveSet",
+  );
+  const eventReserved = events.filter((x) => x.method === "Reserved");
```
Comment (goncer) [inline test/sequential/sequencerStaking.sequencerJoining.test.ts:48]: is not there any other event related with the stake? only tokens.reserve?

**Review by goncer — CHANGES_REQUESTED** [review]

File: test/sequential/sequencerStaking.sequencerJoining.test.ts (hunk 4 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.sequencerJoining.test.ts:48]: Only this one, yes.

File: test/sequential/sequencerStaking.sequencerJoining.test.ts (hunk 2 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.sequencerJoining.test.ts:67]: Done

File: test/sequential/sequencerStaking.sequencerJoining.test.ts (hunk 3 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.sequencerJoining.test.ts:60]: Added

File: test/sequential/sequencerStaking.sequencerJoining.test.ts (hunk 1 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.sequencerJoining.test.ts]: Added functions to filter events and to receive data

File: utils/eventListeners.ts [hunk 5]
```diff
@@ -100,6 +100,15 @@ export function filterEventData(
     .map((event) => event.event.toHuman().data);
 }
 
+export function filterZeroEventData(
```
Comment (goncer) [inline utils/eventListeners.ts:103]: This method filters the event collection by the method param. 
So what does a name like "filterEventsByMethodName" look like?

+ Pz follow linter rules. I guess that this chunk of code does not match the linter rules, 
```


) {
  const filteredResult = result.filter((x) => x.method === method);
  if (filteredResult[0] === undefined){return undefined}else{
  return (JSON.parse(JSON.stringify(filteredResult[0].event.toHuman().data)));}
}
```

File: utils/eventListeners.ts [hunk 6]
```diff
@@ -433,3 +442,21 @@ export async function getEventsAt(blockNo: BN) {
...
+  let isUserJoinedAsSeq: boolean;
+  const eventJoining = filterZeroEventData(events, "SequencerJoinedActiveSet");
+  const eventReserved = filterZeroEventData(events, "Reserved");
+  if (eventJoining !== undefined) {
+    isUserJoinedAsSeq = true;
+  } else {
+    isUserJoinedAsSeq = false;
+  }
+  const userAddress = eventReserved.who;
+  const stakeAmount = (eventReserved.amount).replaceAll(",","");
```
Comment (goncer) [inline utils/eventListeners.ts:456]: use stringToBN instead of this.

File: utils/rollDown/SequencerStaking.ts [hunk 7]
```diff
@@ -24,8 +24,14 @@ export class SequencerStaking {
   static async provideSequencerStaking(
     amount: BN = BN_ZERO,
     chainName: ChainName = "Ethereum",
-    stakeAction: any = "StakeAndJoinActiveSet",
+    isJoinActiveSet = true,
```
Comment (goncer) [inline utils/rollDown/SequencerStaking.ts:27]: rename by: "stakeAndJoin" plz.

**Review by goncer — CHANGES_REQUESTED** [review]

File: utils/eventListeners.ts (hunk 5 above)
Comment (AlexChetverov) [reply utils/eventListeners.ts:103]: Fixed

File: utils/rollDown/SequencerStaking.ts (hunk 7 above)
Comment (AlexChetverov) [reply utils/rollDown/SequencerStaking.ts:27]: Done

**Review by goncer — APPROVED** [review]

**Pull request merged**